src/components/SingleArticle.jsx

```diff
@@ -74,29 +94,31 @@ export default function SingleArticle(){
                 <p className="articleItem"><strong>Topic: </strong> {article.topic}</p>
                 <p className="articleItem"><strong>Article: </strong> {article.body}</p>
                 <img src={article.article_img_url} alt="Article image" className="articleImg" />
-                <p className="articleItem"><strong>Votes: </strong> {article.votes}</p><button className="button" onClick={patchVote}>Vote!</button>
+                <p className="articleItem"><strong>Votes: </strong> {article.votes}</p>
+                <button className="button" onClick={patchVote}>Vote!</button>
                 <p className="articleItem"><small>Posted on </small> {formatDate(dateString)}</p>
                 <p><strong>Comments:</strong> {article.comment_count}</p>
-                <button className="button" onClick={handleClick}>{showComments ? "Hide Comments" : "View Comments"}</button>
-    
-                {showComments && (
-                    <div id="commentsSection">
-                        <div className="commentsContainer">
-                            {comments.length > 0 ? (
-                                comments.map((comment) => (
-                                    <div key={comment.comment_id} className="comment">
-                                        <p><strong>{comment.author}</strong> says:</p>
-                                        <p>{comment.body}</p>
-                                        <p><strong>Votes:</strong> {comment.votes}</p>
-                                        <p><small>Posted on {formatDate(comment.created_at)}</small></p>
-                                    </div>
-                                ))
-                            ) : (
-                                <p>No comments to display.</p>
-                            )}
-                        </div>
-                    </div>
+                <button className="button" onClick={() => setShowComments(!showComments)}>
+                    {showComments ? "Hide Comments" : "View Comments"}
+                </button>
+               
+                 <button className="button" onClick={() => setShowForm(!showForm)}>
+                Have your say
+                    </button>
+                    {showForm && (
+                    <AddComment
+                        author={author}
+                        setAuthor={setAuthor}
+                        body={body}
+                        setBody={setBody}
+                        onSubmit={handleAddComment}
+                     />
+                     )}
+
+                    {successMessage && (
+                    <p style={{ color: 'green' }}>{successMessage}</p>  // Display the success message
                 )}
+                {showComments && <ViewComments article_id={article_id} />}
```

Comment:
I really like this! You could extract it into a component, as described here: https://l2c.northcoders.com/courses/sd-notes/front-end#sectionId=composition-in-react,step=intro